src/amo/reducers/landing.js

```diff
@@ -0,0 +1,36 @@
+import {
+  LANDING_GET,
+  LANDING_LOADED,
+  LANDING_FAILED,
+} from 'core/constants';
+
+
+export const initialState = {
+  featured: { count: 0, results: [] },
+  highlyRated: { count: 0, results: [] },
+  loading: false,
+  popular: { count: 0, results: [] },
+};
+
+export default function landing(state = initialState, action) {
+  const { payload } = action;
+  let newState;
```

Comment:
I find that it's a little cleaner to only define this in the exact switch case that you need it for. You might have tried that and got a lint error because you have to use brackets to declare a new scope:
````javascript
switch (action.type) {
  case LANDING_LOADED: {
    const newState = { ...state, loading: false };
    newState[key] = ...
    return newState;
  } 
}
````